<!-- gl_pose_sampler.launch -->
<launch >
    <arg name="map_yaml_file" default="/media/jhuai/OneTouch/jhuai/data/cane/indoorloc_dataset/level/reloc_data/909/map/map.yaml" />
    <arg name="odometry_file" default="/media/jhuai/OneTouch/jhuai/data/cane/indoorloc_dataset/level/reloc_data/909/test/2024_06_21_17_42_36/odom.txt" />
    <arg name="output_dir" default="/home/jhuai/Desktop/temp" />
    <arg name="open_rviz" default="true" />
    <arg name="node_name" default="gl_pose_sampler_eval" />

    <!-- Input messages -->
    <arg name="map_name" default="/map" /> <!-- nav_msgs::OccupancyGrid -->

    <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_yaml_file)" >
        <remap from="/map" to="$(arg map_name)" />
    </node>

    <node name="mcl_rviz" pkg="rviz" type="rviz" args="-d $(find als_ros)/rviz/als_ros.rviz" if="$(arg open_rviz)" />
    <rosparam command="load" file="$(find als_ros)/config/gl_pose_sampler.yaml" />
    <!-- gl pose sampler node -->
    <node name="$(arg node_name)" pkg="als_ros" type="gl_pose_sampler_eval" output="screen" >
     <param name="odometry_file" value="$(arg odometry_file)" type="string" />
     <param name="output_dir" value="$(arg output_dir)" type="string" />
    </node>
</launch >
